homework_3/task2.c

```diff
@@ -0,0 +1,87 @@
+#include <stdio.h>
+#include <stdlib.h>
+#define arrLn 1000
+void quickSort(int *ar, int first, int last)
+{
+    if (first < last)
+    {
+        int left = first;
+        int right = last;
+        int middle = ar[(first + last) / 2];
+        do
+        {
+            while (ar[left] < middle)
+            {
+                left++;
+            }
+            while (ar[right] > middle)
+            {
+                right--;
+            }
+            if (left <= right)
+            {
+                int temp = ar[left];
+                ar[left] = ar[right];
+                ar[right] = temp;
+                left++;
+                right--;
+            }
+        } while (left <= right);
+
+        quickSort(ar, first, right);
+        quickSort(ar, left, last);
+    }
+}
+
+int binSearch(int *ar, int ln, int target)
+{
+    int left = -1;
+    int right = ln;
+    while (left < right - 1)
+    {
+        int mid = (left + right) / 2;
+        if (ar[mid] > target)
+        {
+           right = mid;
+        }
+        else
+        {
+            left = mid;
+        }
+    }
+    return ar[left] == target;
+}
+
+int main()
+{
```

Comment:
```suggestion
{
        srand(time(NULL));
```
Это способ получить новые псевдослучайные при каждом запуске.
Просто комментарий.